openedx/core/lib/api/view_utils.py

```diff
@@ -401,33 +401,38 @@ def get_course_key(request, course_id=None):
     return CourseKey.from_string(course_id)
 
 
-def verify_course_exists(view_func):
+def verify_course_exists(missing_course_error_message=None):
     """
     A decorator to wrap a view function that takes `course_key` as a parameter.
 
     Raises:
         An API error if the `course_key` is invalid, or if no `CourseOverview` exists for the given key.
     """
-    @wraps(view_func)
-    def wrapped_function(self, request, **kwargs):
-        """
-        Wraps the given view_function.
-        """
-        try:
-            course_key = get_course_key(request, kwargs.get('course_id'))
-        except InvalidKeyError:
-            raise self.api_error(  # lint-amnesty, pylint: disable=raise-missing-from
-                status_code=status.HTTP_404_NOT_FOUND,
-                developer_message='The provided course key cannot be parsed.',
-                error_code='invalid_course_key'
-            )
 
-        if not CourseOverview.course_exists(course_key):
-            raise self.api_error(
-                status_code=status.HTTP_404_NOT_FOUND,
-                developer_message=f"Requested grade for unknown course {str(course_key)}",
-                error_code='course_does_not_exist'
-            )
+    if not missing_course_error_message:
+        missing_course_error_message = "Unknown course {course}"
+
+    def _verify_course_exists(view_func):
+        @wraps(view_func)
+        def wrapped_function(self, request, **kwargs):
+            """
+            Wraps the given view_function.
+            """
+            try:
+                course_key = get_course_key(request, kwargs.get('course_id'))
+            except InvalidKeyError:
+                raise self.api_error(  # lint-amnesty, pylint: disable=raise-missing-from
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    developer_message='The provided course key cannot be parsed.',
+                    error_code='invalid_course_key'
+                )
+            if not CourseOverview.course_exists(course_key):
+                raise self.api_error(
+                    status_code=status.HTTP_404_NOT_FOUND,
+                    developer_message=missing_course_error_message.format(course=str(course_key)),
```

Comment:
I know it does but I have seen this convention almost everywhere where usage keys are involved. I'm not sure why it is this way. Perhaps it is a historical reason? 

In any case, I tend to cast it to str when I use it as such just to be explicit and in case there is a good reason why it's done all over the place.